docs/theme/js/switcher.js

```diff
@@ -87,10 +87,36 @@
       return ''
     }
   
+    function nthIndex(str, pat, n){
+      var L= str.length, i= -1;
```

Comment:
Looks like you copied those functions from stackoverflow but imho that’s pretty unreadable. Iterating over regexp matches which already support that natively here and just calling `.length` on the result for `count` would be much cleaner imho.

Reply:
How would you implement `nthIndex` using RegExp matches? I can see it for `count`:

```
  function count(str, pat){
      var re = new RegExp(pat, "g");
      return (str.match(re) || []).length;
    }
```

Reply:
Iterate `n` times over the result of `exec`

Reply:
Interesting, I didn't know about the stateful use of `exec`. This is what I came up with. Is it easier to understand in your opinion?

```
    function nthIndex(str, pat, n){
      var re = new RegExp(pat, "g");
      var m = null;
      for(var i = 0; i < n; i++)
        m = re.exec(str);
      return m ? re.lastIndex : -1;
    }
```

Reply:
I like it, thx!